test/val/val_fixtures.h

```diff
@@ -18,32 +18,65 @@
 #define LIBSPIRV_TEST_VALIDATE_FIXTURES_H_
 
 #include "unit_spirv.h"
+#include "test_fixture.h"
+
+#include <algorithm>
+#include <string>
+#include <vector>
 
 namespace spvtest {
 
 template <typename T>
 class ValidateBase : public ::testing::Test,
                      public ::testing::WithParamInterface<T> {
  public:
-  ValidateBase();
+  ValidateBase() : diagnostic_() {}
 
-  virtual void TearDown();
+  virtual void TearDown() {
+    if (diagnostic_) {
+      spvDiagnosticPrint(diagnostic_);
+    }
+    spvDiagnosticDestroy(diagnostic_);
+  }
 
   // Returns the a spv_const_binary struct
-  spv_const_binary get_const_binary();
 
   void CompileSuccessfully(std::string code,
-                           spv_target_env env = SPV_ENV_UNIVERSAL_1_0);
+                           spv_target_env env = SPV_ENV_UNIVERSAL_1_0) {
+    spv_diagnostic diagnostic = nullptr;
+    spv_binary binary;
+    ASSERT_EQ(SPV_SUCCESS,
+              spvTextToBinary(ScopedContext(env).context, code.c_str(),
+                              code.size(), &binary, &diagnostic))
+        << "ERROR: " << diagnostic->error
+        << "\nSPIR-V could not be compiled into binary:\n"
+        << code;
+
+    words_.reserve(binary->wordCount);
+    std::copy(binary->code, binary->code + binary->wordCount,
+         std::back_inserter(words_));
+  }
+
+  void SetBinary(std::vector<uint32_t> &&words) {
+    words_ = move(words);
+  }
 
   // Performs validation on the SPIR-V code and compares the result of the
   // spvValidate function
-  spv_result_t ValidateInstructions(spv_target_env env = SPV_ENV_UNIVERSAL_1_0);
+  spv_result_t ValidateInstructions(
+      spv_target_env env = SPV_ENV_UNIVERSAL_1_0) {
+    return spvValidateBinary(ScopedContext(env).context,
+                       words_.data(), words_.size(),
+                       &diagnostic_);
+  }
 
-  std::string getDiagnosticString();
-  spv_position_t getErrorPosition();
+  std::string getDiagnosticString() {
+    //std::cout << diagnostic_->error << std::endl;
+    return std::string(diagnostic_->error); }
+  spv_position_t getErrorPosition() { return diagnostic_->position; }
 
-  spv_binary binary_;
   spv_diagnostic diagnostic_;
+  std::vector<uint32_t> words_;
```

Comment:
I see that `spv_binary binary_` has been replaced with `std::vector<uint32_t> words_`.
And also val_fixtures.cpp has been fully removed (and some functionality has been moved to val_fixtures.h). Is there a particular reason for these changes?

I think this part of the CL may be either removed (not necessary) or separated out from the type checking code.